### PR TITLE
feat(network-columns-resize)/added resizable columns in network panel

### PR DESCRIPTION
**Pull request by sachin-thakur-bruno**
### Description

The network tab table columns in dev tools have static columns width. Extending the existing table columns to be resizable and persisting the user preference in local storage.

[JIRA](https://usebruno.atlassian.net/browse/BRU-3584)

Adds drag-to-resize column support to the Devtools Network Tab.

- **`useResizableColumns` hook** — new reusable hook that manages column widths in `px`, measures the container via `ResizeObserver`, and scales all columns proportionally on container resize. Adjacent-column resize (left grows ↔ right shrinks) with a hard stop at `MIN_COL_WIDTH = 60px`.
- **`NetworkTab`** — wired up to the hook; separator divs span the full table height as drag handles.
- **`utils.js`** — cleaned up; grid/separator helpers moved into the hook.

### Behavior

- Total width always equals container width — no horizontal scroll.
- Dragging only affects the two columns around the grabbed separator.
- Columns scale proportionally when the panel is resized.
- Separator styling matches the existing Request Details drag handle (`sidebar.dragbar.border` / `sidebar.dragbar.activeBorder`).
- 
### Network Tab: Persist Sort Config
 
**`NetworkTab/index.js`**
 
Sort column and direction were stored in local component state (`useState`), so they reset every time the devtools panel was closed. Replaced with `usePersistedState` keyed to `devtools-network-sort`.

### Bug Fix: Details Panel Dismissed on Tab Switch
 
**`slices/logs.js`**
 
`setActiveTab` was unconditionally nulling `selectedRequest` whenever the user navigated away from the Network tab. This meant the request details panel was always lost on tab switches, even though it should survive.
 
### Request Details Panel: Migrate Width from Redux to localStorage
 
**`Console/index.js`** · **`slices/logs.js`**
 
Panel width was stored in Redux state (`requestDetailsPanelWidth`) and persisted via a dedicated action (`updateRequestDetailsPanelWidth`). Since this is purely UI preference with no cross-component dependencies, it's been moved to `usePersistedState` keyed to `devtools-details-panel-width`.
 
Removed from Redux:
- `requestDetailsPanelWidth` initial state field
- `updateRequestDetailsPanelWidth` reducer and export

### Tests

18 tests covering initial state, container measurement, proportional rescaling, adjacent-column drag, hard-stop conservation, and unmount cleanup.

https://github.com/user-attachments/assets/ba9388c4-94da-4b81-a1ab-74ddac73e2e2

#### Contribution Checklist:

- [x] **I've used AI significantly to create this pull request**
- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have added screenshots or gifs to help explain the change if applicable.**
- [x] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [x] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

#### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

## Summary by CodeRabbit

* **New Features**
  * Added draggable column separators to resize Network tab columns.
* **Improvements**
  * Enhanced Network tab sorting (method, status, domain/path, time/duration, size) with preferences persisted.
  * Refreshed Network tab layout/styling for a more consistent grid/table feel, including better scrolling, truncation, and resizing visuals.
  * Persist the Console details panel width locally.
* **Tests**
  * Added/expanded coverage for column resizing (including min-width edge cases) and NetworkTab persistence behavior.

### CODE REVIEW

File: packages/bruno-app/src/hooks/useResizableColumns/index.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  }
+  return positions;
+};
+
+const scaleWidthsToTotal = (widths, targetTotal, minColWidth) => {
+  const currentTotal = widths.reduce((s, w) => s + w, 0);
+  const factor = targetTotal / currentTotal;
+  const next = widths.slice(0, -1).map((w) => Math.max(minColWidth, Math.round(w * factor)));
+  const last = Math.max(minColWidth, targetTotal - next.reduce((s, w) => s + w, 0));
+  return [...next, last];
```
[the remark this answers is not in the document]
Comment (sachin-thakur-bruno) [reply packages/bruno-app/src/hooks/useResizableColumns/index.js:51]: This is more of a defensive correctness fix. But since hook is reusable and a future caller might have more columns or a smaller container, added a fall back to equal distribution so the total stays exact (columns clip instead of overflow). Also added a test around it

File: packages/bruno-app/src/hooks/useResizableColumns/index.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+    const onMouseUp = () => {
+      setResizingIdx(null);
+      document.removeEventListener('mousemove', onMouseMove);
+      document.removeEventListener('mouseup', onMouseUp);
+    };
+
+    document.addEventListener('mousemove', onMouseMove);
+    document.addEventListener('mouseup', onMouseUp);
+  }, [colWidths, minColWidth]);
```
[the remark this answers is not in the document]
Comment (sachin-thakur-bruno) [reply packages/bruno-app/src/hooks/useResizableColumns/index.js:152]: added a ref and clean up

File: packages/bruno-app/src/components/Devtools/Console/NetworkTab/index.js [hunk 3]
```diff
@@ -219,22 +223,24 @@ const NetworkTab = () => {
...
-              ) : null
-            )}
+            {separatorPositions.map((left, i) => (
+              <div
+                key={i}
+                className={`col-separator${resizingIdx === i ? ' resizing' : ''}`}
+                style={{ left }}
+                onMouseDown={(e) => handleResizeStart(e, i)}
+              />
+            ))}
```
[the remark this answers is not in the document]
Comment (sachin-thakur-bruno) [reply packages/bruno-app/src/components/Devtools/Console/NetworkTab/index.js:248]: added the data test id

**Review by sid-bruno — APPROVED** [review]
